CareEvolution/CE-API-Demo-Android
# Pull in the latest libraries, migrate to AndroidX, and update documentation

# Overview

* Updated all libraries and switched to a Gradle [version catalog](https://docs.gradle.org/current/userguide/platforms.html).
* Replaced [ButterKnife](https://jakewharton.github.io/butterknife/) with [View Binding](https://developer.android.com/topic/libraries/view-binding).
* Switched to a Material theme with palette from the [Material Theme Builder](https://m3.material.io/theme-builder).
* Bumped source compatibility to Java 11.
* Fixed a pull-to-refresh bug on the patient list screen.
* Updated the documentation and corrected broken links.

# Testing

First, test the documentation by configuring a new client app:

1. Follow the [SMART Client Registration](https://fhir.docs.careevolution.com/overview/tutorials/authentication.html#smart-client-registration) guide register a new client app.
2. Set `OAuth_clientId` and  `OAuth_clientSecret` with values from the client registration above.  The client secret is only shown once and should be stored in a password manager.  The client ID is a UUID &mdash; _not_ the display name assigned to the client.

Second, run through the main features:

* login & logout
* canceled / denied authentication
* pull-to-refresh

# Security

Double-check `config.xml` and confirm that it does not contain client IDs and access keys.
